### PR TITLE
Define FriendlyName for the entities, based on deck/state/types.go

### PR DESCRIPTION
**Pull request by mmorel-35**
This feature add identifier methods to entities to allow to print names or ids depending on which one is available.
It will offer a clean solution for https://github.com/Kong/deck/issues/417

### CODE REVIEW

File: kong/identifiers.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	}
+	return *r1.ID
+}
+
+// Identifier returns a composite ID base on Role ID, workspace, and endpoint
+func (r1 *RBACEndpointPermission) Identifier() string {
+	if r1.Endpoint != nil {
+		return fmt.Sprintf("%s-%s-%s", *r1.Role.ID, *r1.Workspace, *r1.Endpoint)
+	}
+	return *r1.Endpoint
```
Comment (mflendrich) [inline kong/identifiers.go:90]: `Endpoint` is nil here and it cannot be dereferenced.

File: kong/identifiers.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package kong
+
+import "fmt"
+
+// Identifier returns the endpoint key name or ID.
+func (s1 *Service) Identifier() string {
```
Comment (mflendrich) [inline kong/identifiers.go:6]: Please call this function `FriendlyName` to better describe its purpose.

File: kong/identifiers.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package kong
```
Comment (mflendrich) [inline kong/identifiers.go:1]: Please implement methods alongside type definitions, in the same file.

File: kong/identifiers.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+import "fmt"
+
+// Identifier returns the endpoint key name or ID.
+func (s1 *Service) Identifier() string {
+	if s1.Name != nil {
+		return *s1.Name
+	}
+	return *s1.ID
```
Comment (mflendrich) [inline kong/identifiers.go:10]: For all types please only dereference ID if it's non-nil.

If none of the checked fields is non-nil, return `""` as the friendly name.

File: kong/identifiers.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+func (r1 *RBACRole) Identifier() string {
+	if r1.Name != nil {
+		return *r1.Name
+	}
+	return *r1.ID
+}
+
+// Identifier returns a composite ID base on Role ID, workspace, and endpoint
+func (r1 *RBACEndpointPermission) Identifier() string {
+	if r1.Endpoint != nil {
```
Comment (mflendrich) [inline kong/identifiers.go:87]: This needs to check all the dereferenced fields.

File: kong/identifiers.go (hunk 2 above)
Comment (mflendrich) [inline kong/identifiers.go:6]: Please drop the `1`s from all the receiver names.

Comment (mmorel-35) [reply kong/identifiers.go:6]: There is already Identifier or Console which behave the same way in deck, are you sure about this name?

Comment (mmorel-35) [reply kong/identifiers.go:6]: It does, I'll take care of this this weekend

**Comment by mmorel-35** [comment]
Hi @mflendrich, anything left to do here?

**Review by mflendrich — APPROVED** [review]

**Comment by mmorel-35** [comment]
Any idea when is planned the next release for go-kong?

**Pull request merged**